### PR TITLE
Improved probe implementation, error logging, documentation and naming of srf02_i2c driver.

### PR DESCRIPTION
**Pull request by cehberlin**
- Probe is now using the version register of the sensor
 - Removed/renamed many constants and strings that have been copied from the maxbotics driver origin
 - Improved error logging in case something is not working as expected

### CODE REVIEW

File: src/drivers/srf02_i2c/srf02_i2c.cpp [hunk 1]
```diff
@@ -733,21 +760,25 @@ start(uint8_t rotation)
 	g_dev = new SRF02_I2C(rotation, SRF02_I2C_BUS);
 
 	if (g_dev == nullptr) {
+		err(1,"SRF02_I2C constructor failed");
```
Comment (dagar) [inline src/drivers/srf02_i2c/srf02_i2c.cpp]: Can you change these to PX4_ERR, followed by return? This err actually exits, which is fine on NuttX, but when running on Linux exits the entire running firmware.

**Comment by dagar** [comment]
@cehberlin can you rebase to resolve the conflicts, then please CI output to see the minor build and formatting errors.

**Comment by cehberlin** [comment]
@dagar I thought you expected me to rebase to master, which I did. However, I am right now quite confused why the driver and my other do not exist anymore? How to mirgrate my changes if the driver was removed? Can you give me some background how this should be done?

**Comment by dagar** [comment]
@cehberlin all the distance sensors were moved to src/drivers/distance_sensor. This allows a board to include them all easily. Otherwise the actual code shouldn't have changed at all.

**Pull request closed without merging**